build/accelerate_launch.py

```diff
@@ -159,11 +170,30 @@ def main():
             for _, dirs, _ in os.walk(output_dir, topdown=False):
                 for name in dirs:
                     if "checkpoint-" in name.lower():
-                        post_process_vLLM_adapters_new_tokens(
-                            os.path.join(output_dir, name),
-                            os.path.join(output_dir, name),
-                            num_added_tokens,
+                        checkpoint_dir = os.path.join(output_dir, name)
+                        if os.path.exists(
+                            os.path.join(checkpoint_dir, "adapter_model.safetensors")
+                        ):
+                            post_process_vLLM_adapters_new_tokens(
+                                checkpoint_dir,
+                                checkpoint_dir,
+                                num_added_tokens,
+                            )
+
+                        # In case of ScatterMoE LoRa
+                        hf_converted_checkpoint = os.path.join(
+                            checkpoint_dir, "hf_converted_checkpoint"
                         )
+                        if os.path.exists(
+                            os.path.join(
+                                hf_converted_checkpoint, "adapter_model.safetensors"
+                            )
+                        ):
+                            post_process_vLLM_adapters_new_tokens(
+                                hf_converted_checkpoint,
+                                hf_converted_checkpoint,
+                                num_added_tokens,
+                            )
```

Comment:
We can simplify this code as in you can prepare checkpoint_dir with  `output_dir, name` or `output_dir, name, hf_converted_checkpoint` by checking if `hf_converted_checkpoint` exists (`os.path.exists`) and then move to below code with the prepared checkpoint_dir

```
                        if os.path.exists(
                            os.path.join(checkpoint_dir, "adapter_model.safetensors")
                        ):
                            post_process_vLLM_adapters_new_tokens(
                                checkpoint_dir,
                                checkpoint_dir,
                                num_added_tokens,
                            )
```